src/components/Form/formFunctions.ts

```diff
@@ -120,6 +120,59 @@ export const onChangeRadioGroup = (
     };
 };
 
+export const valueOnChangeRadioGroup = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string
+): selectChangeEventHandler => {
+    return (event) => {
+        const input = event.target.value;
+        fieldSetter(key, input);
+        errorSetter(key, Errors.none);
+    };
+};
+
+export const onChangeDate = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string,
+    value: Date | null
+): void => {
+    if (value === null) {
+        return;
+    }
+    const input = new Date(value);
+    const date = {
+        year: input?.getFullYear(),
+        month: input?.getMonth(),
+        day: input?.getDate(),
+    };
+    fieldSetter(key, value);
+    !(date.year && date.month && date.day)
+        ? errorSetter(key, Errors.invalid)
+        : errorSetter(key, Errors.none);
+};
+
+export const onChangeTime = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string,
+    value: Date | null
+): void => {
+    if (value === null) {
+        fieldSetter(key, null);
+        return;
+    }
```

Comment:
May want to change the condition `value === null` to `(value === null || isNaN(Date.parse(value))` as it will prevent invalid `value` variables from being converted to a `Date`. Then, after the `if` statement, you now know that the corresponding `Date` is valid and so you can simplify some of the code below

Reply:
Changed!